### PR TITLE
Avoid splitting log line across two files

### PR DESCRIPTION
**Pull request by dadgar**
Fixes https://github.com/hashicorp/nomad/issues/1048

### CODE REVIEW

File: client/driver/logging/rotator.go [hunk 1]
```diff
@@ -70,28 +83,52 @@ func NewFileRotator(path string, baseFile string, maxFiles int,
...
+		remainingToWrite := int64(len(p[n:]))
+
+		// Check if we are near the end of the file. If we are we attempt to
+		// avoid a log line being split between two files.
+		var nw int
+		if (remainingSpace - lineScanLimit) < remainingToWrite {
+			// Scan for new line and if the data up to new line fits in current
+			// file, write to buffer
+			idx := bytes.IndexByte(p[n:], newLineDelimiter)
+			if idx >= 0 && (remainingSpace-int64(idx)-1) >= 0 {
```
Comment (preetapan) [inline client/driver/logging/rotator.go]: Does this conditional work right if the first char in p is a new line, i.e idx=0? 
IIUC, It would write just the new line char to the buffer and then get back to the for loop. 

Might want to make a unit test for such edge cases though?

File: client/driver/logging/rotator_test.go [hunk 2]
```diff
@@ -168,6 +169,72 @@ func TestFileRotator_RotateFiles(t *testing.T) {
...
+	if err != nil {
+		t.Fatalf("test setup err: %v", err)
+	}
+
+	// We will write two times:
+	// 1st: Write with new lines spanning two files
+	// 2nd: Write long string with no new lines
+	expectations := [][]byte{
+		[]byte("ab\n"),
+		[]byte("cdef\n"),
```
Comment (preetapan) [inline client/driver/logging/rotator_test.go]: would suggest adding test cases with new line in the beginning, and middle of string

Comment (dadgar) [reply client/driver/logging/rotator_test.go]: str1 captures the middle but I can add one for the beginning

File: client/driver/logging/rotator.go (hunk 1 above)
Comment (dadgar) [reply client/driver/logging/rotator.go]: Yeah that is what the minus 1 is for. Since a write of p[n:n] actually writes 1 byte so I have to check if there is room for whatever index + 1
